src/serde_helpers.rs

```diff
@@ -374,20 +374,13 @@ pub mod uuid_as_binary {
         D: Deserializer<'de>,
     {
         let binary = Binary::deserialize(deserializer)?;
-        if binary.subtype == BinarySubtype::Uuid {
-            if binary.bytes.len() == 16 {
-                let mut bytes = [0u8; 16];
-                bytes.copy_from_slice(&binary.bytes);
-                Ok(Uuid::from_bytes(bytes))
-            } else {
-                Err(de::Error::custom(
-                    "cannot convert Binary to Uuid: incorrect bytes length",
-                ))
-            }
+        let res = binary.to_uuid();
```

Comment:
I'm not thrilled with this unpacking and then repacking of the result of to_uuid but returning the result directly doesn't type check. I'm not sure if it's possible to constrain `D::Error` to be of type `crate::de::Error` but it doesn't compile when I directly attempt the constraint.

Reply:
All `Error` types implement `Display` (see trait requirement [here](https://doc.rust-lang.org/std/error/trait.Error.html)), so we can avoid destructuring and just do the following:
```rust
binary.to_uuid().map_err(de::Error::custom)?
```
And this works because [`serde::de::Error::custom`](https://docs.serde.rs/serde/de/trait.Error.html#tymethod.custom) accepts any `T: Display`. This will probably clone the error, but the error path isn't the "hot path" so it's not really a concern.